nefertari/elasticsearch.py

```diff
@@ -245,6 +242,74 @@ def get_document_proxies_by_type(cls, doc_type):
         raise UnknownDocumentProxiesTypeError('You have no proxy for this %s document type' % doc_type)
 
 
+class BoundedTransaction(type):
+
+    def __call__(cls, *args, **kwargs):
+        import transaction
+        current_transaction = transaction.get()
+        es_action_registry = ESActionRegistry()
+        es_action = super(BoundedTransaction, cls).__call__(*args, **kwargs)
+        es_action_registry.subscribe_on_after_commit(current_transaction, es_action)
+        return es_action
+
+
+class ESActionRegistry(metaclass=SingletonMeta):
+
+    def __init__(self):
+        self.registry = {}
+
+    def subscribe_on_after_commit(self, transaction, es_action):
+        if transaction in self.registry:
+            self.registry[transaction].append(es_action)
```

Comment:
Python interpreter implicitly calls __hash__ method from an instance. I can explicilty call this method, because we don't need transaction instance.